### ISSUE TITLE
Help getting to 1.0

### ISSUE THREAD

**Issue by benmccann** [sveltejs/kit#2100]
### Describe the problem

We're actively working towards getting SvelteKit 1.0 out! But there's a lot to do. Just reviewing incoming PRs, prioritizing issues for 1.0, suggesting solutions for important issues, etc. is nearly a fulltime job.

We've knocked out over 500 issues that were on the 1.0 milestone, but there's still a couple dozen left

### Describe the proposed solution

If you'd like to help, please consider working on any of these issues:

https://github.com/sveltejs/kit/issues?q=is%3Aopen+is%3Aissue+milestone%3A1.0

The most straightforward items have been labeled with "help wanted"

For those, or any other tickets, we're happy to help get to a consensus on what the solution should be if it's not clear. Feel free to ping the ticket or find me on Discord under this same username. Take a look at [the contributor's guide](https://github.com/sveltejs/kit/blob/master/CONTRIBUTING.md) to get started, but feel free to ask if additional pointers would be helpful

### Alternatives considered

The maintainers do everything and 1.0 never happens :smile: 

### Importance

blocking an upgrade

**Comment by kjaonline** [comment]
How can a noob like me contribute?

**Comment by denny64** [comment]
Is there a backlog of things to do and their priorities? Happy to contribute wherever I can 😊

**Comment by benmccann** [comment]
> How can a noob like me contribute?

Helping to answer people's questions on Discord or StackOverflow can be a good way to contribute as a beginner since you'll often learn a lot in the process

> Is there a backlog of things to do and their priorities?

https://github.com/sveltejs/kit/issues basically. A few ways to find important things to work on:
- filter by the 1.0 milestone
- filter by p1 label
- sort by thumbs up

**Comment by joglomedia** [comment]
Willing to help, but I am still n00b :grin:

**Comment by lamualfa** [comment]
For anyone who feel **noob**, there are many other things you can do. For example like give a little donation to Svelte team in https://opencollective.com/svelte. Whatever donation you give, it will help the team to maintain many awesome Svelte projects now and future. 

❤️❤️❤️

**Comment by antony** [comment]
n00bs - we all benefit from writing tests. Good tests are always welcome.

If you think you are too n00b to contribute (you're not), then add tests, or write tests for the feature you want to add, before you add it! Start small, and learn the codebase that way.

If you need help, or want to discuss how to contribute, it's better discussed in the svelte kit contributing channel on [discord](https://svelte.dev/chat). It's much easier to get a realtime answer there and chat to us and other contributors directly.

**Comment by vwkd** [comment]
I feel like documentation should be improved before declaring 1.0. Currently there is just so much undocumented or poorly documented. I wish I could help, but there is so much black magic going on under the hood. I feel like a maintainer sitting down and typing out the rough picture would really help.

---

Seeing that they actually raised a decent amount of money raises (pun) the question, why they don't put some people on it full time. It seems to have been the better part of the year since SvelteKit was announced. There seemed to be quite some hype last year. I wonder where it went... Now what they have some money, I wish @Rich-Harris would actually use it. Hire some people, improve the product, get more funding with an improved product, repeat.

It seems like @benmccann is currently the only brave one voluntarily fighting the flood of ever growing issues! I'm having the impression both Svelte and SvelteKit aren't well organised. PRs hang forever, because it's not clear who should make decisions. It really seems like the initial maintainer has never really freed this project from his personal side gig to a grown up project. In its current state it can't seriously compete with the bigger frameworks, even though IMO it has potential to be better than any of them.

**Comment by benmccann** [comment]
It's unclear to me what you find lacking in the documentation. PRs or issues to improve docs are welcome. However, please do search the backlog for existing issues before filing new ones. I see that you filed three duplicates of what are all essentially the same already existing issue in the past few hours.

Svelte currently has ~$40k annual budget. That's not enough to support anyone working on Svelte full-time. That being said, we are discussing options for using the funds to get someone paid working on it part-time.

The PRs for this project are actively reviewed. Almost all the PRs more than a week old that haven't been merged are blocked on the PR submitter responding to our questions or comments and have been labeled as such. I may be the person most visible on this project, but I am frequently discussing APIs and open PRs with the other maintainers behind the scenes and getting their help to fix issues.

Please recognize that the Svelte projects are maintained purely by volunteers. You are welcome to help! Please send PRs for things you think could be improved!

**Comment by vwkd** [comment]
Thanks @benmccann for taking a look at my latest three issues. Not sure where to put this, since the three issues are each a bad place. But would you mind explaining, maybe somewhere else like over Discord, how they relate to the apparent original #1540 with title "Define default meta tags, title etc. and have the option to override it"? None of the three have something to do with default titles. The only thing I can see that they have in common is the word "title" in the title.

I'm quite surprised how quickly you were able to figure out those three were all duplicates of one underlying issue. It took me several hours to even wrap my head around these issues, clearly documenting them using a screencast in as well as a working reproduction example. Meanwhile it took you 2 (!) minutes (timestamps 11:23, 11:24, and 11:25 of closing comment) to understand all of them and see they are apparently duplicates of #1540.

It seems like this project has bigger problems than I expected. Closing issues to get the count down. Also marking every comment (including of course this one) off-topic.

I'm sorry for having bothered you. I thank you for inviting me to help. I had seriously considered it. But I feel like this place isn't welcoming me. I won't bother you any further. Good luck!

**Comment by rchrdnsh** [comment]
I contributed on open collective XD

**Comment by janat08** [comment]
Issuehunt lets you post bounties, and I see like 80k in opencollective. In the 3rd world paying bills is a concern as a developer.

**Comment by benmccann** [comment]
I've heard that issuehunt is non-responsive and the money there never gets paid out.

We are looking to spend the OpenCollective money. The hard part thus far has been finding someone both qualified and willing

**Comment by janat08** [comment]
I'm by in large both unqualified and somewhat unwilling, but definitely willing in so far as pay amounts go. Can I go and try submit an expense? There was that issue with boiler plate templates being misconfigured.

**Comment by benmccann** [comment]
It would take a lot of work to review PRs from a wide range of submitters who have various levels of familiarity with the codebase and would be extra overhead to individually negotiate each ticket. We'll be looking to use the OpenCollective funds with someone we've already had a long-term relationship with as a contributor instead.

**Comment by antony** [comment]
For future reference @janat08, we would also be looking to determine the value/price of the work, and also the specific approach prior to that work being done, not retrospectively being billed for work that already took place. Not all issues are something which would attract a bounty, which is another reason that price discussions should take place prior to any work taking place.

**Comment by janat08** [comment]
So for example https://github.com/sveltejs/kit/issues/4135at5$ an hour as I have no clue what's up with svelte and where this will even go? I presume it's just some problem with adapters. It's a p1 issue and I have poor time management so the value of work equals out.

**Comment by benmccann** [comment]
@janat08 as I said in my last comment, we are only looking to hire people who have demonstrated an ability to contribute to Svelte/SvelteKit. We will not be paying you to contribute in any fashion. Please stop asking

**Comment by itswadesh** [comment]
As a noob, I am contributing this project by making a complete open source ecommerce store front using Sveltekit named svelte-commerce. This project has proved that Sveltekit is ready for production use and performs much much better than any other Javascript library out there. So that developers will feel confident and start advocating towards it while making tech-stack decisions.

**Comment by lordvcs** [comment]
Can I suggest something like what nextjs does for their issues[https://github.com/vercel/next.js/issues], on visiting the nextjs issues section you are greeted with this header section,
![image](https://user-images.githubusercontent.com/6374881/179345062-b435186b-011f-4df3-be25-227058353676.png)
Clicking on the [we've collected some good first issues for you](https://github.com/vercel/next.js/contribute) takes you to the list of issues which have been tagged as [good first issue](https://github.com/vercel/next.js/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22). This can potentially bring in a lot of contributors becuase getting started is the hard part, especially for so-called noobs(like me).

**Comment by dlicois** [comment]
> There's a bit of a [guide to the codebase](https://github.com/sveltejs/kit#code-structure) on the readme

Hello, 
you should update this link to point to https://github.com/sveltejs/kit/blob/master/CONTRIBUTING.md#code-structure (or rephrase the post to link to the CONTRIBUTING file), since this issue is in high visibility.

**Comment by IRod22** [comment]
Congrats on almost reaching v1! :tada: I would also recommend changing the warning on the docs as follows (or I can submit a PR for the edit):

```diff
- SvelteKit is in early development, and some things may change before we hit version 1.0. . . .
+ SvelteKit is in its release candidate phase. . . .
```

**Comment by benmccann** [comment]
I'm going to close this as we don't have all that many issues left anymore
